File: src/main/java/pe/nanamochi/banchus/entities/SlotTeam.java
package pe.nanamochi.banchus.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum SlotTeam {
  NEUTRAL(0),
  BLUE(1),
  RED(2);

  private final int value;

  public static SlotTeam fromValue(int value) {
    for (SlotTeam team : values()) {
      if (team.value == value) {
        return team;
      }
    }
    throw new IllegalArgumentException("Unknown team value: " + value);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/components/SlotTeam.kt
package pe.nanamochi.banchus.components

enum class SlotTeam(val value: Int) {
    NEUTRAL(0),
    BLUE(1),
    RED(2);

    companion object {
        fun fromValue(value: Int): SlotTeam = entries.find { it.value == value } ?: NEUTRAL
    }
}
